src/build/app/mod.rs

```diff
@@ -1481,40 +1470,160 @@ impl<'b> App<'b> {
+                assert!(
+                    self.args.args.iter().filter(|x| x.long == Some(l)).count() < 2,
+                    "Argument long must be unique\n\n\t'--{}' is already in use",
+                    l
+                );
+            }
+
+            // Short conflicts
+            if let Some(s) = arg.short {
+                assert!(
+                    self.args.args.iter().filter(|x| x.short == Some(s)).count() < 2,
+                    "Argument short must be unique\n\n\t'-{}' is already in use",
+                    s
+                );
+            }
+
+            // Index conflicts
+            if let Some(idx) = arg.index {
+                assert!(
+                    positionals!(self).fold(0, |acc, p| if p.index == Some(idx) {
+                        acc + 1
+                    } else {
+                        acc
+                    }) < 2,
+                    "Argument '{}' has the same index as another positional argument\n\n\t \
+                    Use Arg::setting(ArgSettings::MultipleValues) to allow one \
+                    positional argument to take multiple values",
+                    arg.name
+                );
+            }
+
+            // requires, r_if, r_unless
+            if let Some(reqs) = &arg.requires {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == req.1)
+                            || self.groups.iter().any(|x| x.id == req.1),
+                        "Argument or group specified in 'requires*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            if let Some(reqs) = &arg.r_ifs {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == req.0)
+                            || self.groups.iter().any(|x| x.id == req.0),
+                        "Argument or group specified in 'required_if*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            if let Some(reqs) = &arg.r_unless {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == *req)
+                            || self.groups.iter().any(|x| x.id == *req),
+                        "Argument or group specified in 'required_unless*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            // blacklist
+            if let Some(reqs) = &arg.blacklist {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == *req)
+                            || self.groups.iter().any(|x| x.id == *req),
+                        "Argument or group specified in 'conflicts_with*' for '{}' does not exist",
```

Comment:
Seeing the name of the group may be useful, don't you think?

Reply:
Can't do it because `pub blacklist: Option<Vec<Id>>,` which means the argument name is `Id` and not string